src/frontend/js/components/DashBoard/Navigator/index.tsx

```diff
@@ -0,0 +1,48 @@
+import { NavLink, useRoutes } from 'react-router-dom';
+import { ReactNode } from 'react';
+import { defineMessages, FormattedMessage } from 'react-intl';
+import useDashboardRoutes from '../hooks/useDashboardRoutes';
+import { Spinner } from '../../Spinner';
+
+export const messages = defineMessages({
+  loadingNavigator: {
+    id: 'components.Dashboard.Navigator.loadingInitial',
+    defaultMessage: 'Loading navigator',
+    description:
+      'Accessible text for the initial loading spinner displayed when navigator is fetching',
+  },
+});
+
+const DashBoardNavigator = () => {
+  const { intlDashboardRoutes } = useDashboardRoutes();
+  const routesNode = useRoutes(intlDashboardRoutes);
+
+  if (!intlDashboardRoutes) {
+    return (
+      <Spinner aria-labelledby="loading-routes">
+        <span id="loading-course">
+          <FormattedMessage {...messages.loadingNavigator} />
+        </span>
+      </Spinner>
+    );
+  }
```

Comment:
Currently, the `useDashboardRoutes` is not async right? So I do not understand when the spinner will be displayed.

Furthermore, I think we should raise an error if `dashboardRoutes` are undefined.